osquery/core/tables.cpp

```diff
@@ -91,35 +101,42 @@ void TablePlugin::setContextFromRequest(const PluginRequest& request,
 Status TablePlugin::call(const PluginRequest& request,
                          PluginResponse& response) {
   response.clear();
-  // TablePlugin API calling requires an action.
+
   if (request.count("action") == 0) {
     return Status(1, "Table plugins must include a request action");
   }
 
-  if (request.at("action") == "generate") {
-    // The "generate" action runs the table implementation using a PluginRequest
-    // with optional serialized QueryContext and returns the QueryData results
-    // as the PluginRequest data.
+  const auto& action = request.at("action");
 
-    // Create a fake table implementation for caching.
-    QueryContext context;
-    if (request.count("context") > 0) {
-      setContextFromRequest(request, context);
-    }
+  if (action == "generate") {
+    auto context = queryContextFromRequest(request);
     response = generate(context);
-  } else if (request.at("action") == "columns") {
-    // The "columns" action returns a PluginRequest filled with column
-    // information such as name and type.
+
+  } else if (action == "delete") {
+    auto context = queryContextFromRequest(request);
+    response = delete_(context, request);
+
+  } else if (action == "insert") {
+    auto context = queryContextFromRequest(request);
+    response = insert(context, request);
+
+  } else if (action == "update") {
+    auto context = queryContextFromRequest(request);
+    response = update(context, request);
+
+  } else if (action == "columns") {
+    static_cast<void>(request);
```

Comment:
What does this do?

Reply:
This is just to make static analysis (and warnings with -Wunused) happy; I kind of keep adding it due to habits for other projects! I can see it's not applicable here, I'll remove it!